pubsub/azurepubsub/azurepubsub.go

```diff
@@ -0,0 +1,438 @@
+	s, err := openSubscription(ctx, topicName, subscriptionName, connString, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return pubsub.NewSubscription(s, nil), nil
+}
+
+// openSubscription returns a driver.Subscription.
+func openSubscription(ctx context.Context, topicName string, subscriptionName string, connectionString string, opts *SubscriptionOptions) (driver.Subscription, error) {
+	ns, err := getSbNamespace(connectionString)
+	if err != nil{
+		return nil, err
+	}
+
+	defaultTimeout := listenerTimeout
+	if opts != nil {
+		if opts.ListenerTimeout > 0 {
+			defaultTimeout = opts.ListenerTimeout
+		}
+	}
+
+	return &subscription {		
+		name : subscriptionName,
+		topicName: topicName,		
+		ns: ns,
+		opts : &SubscriptionOptions {
+			ListenerTimeout: defaultTimeout,
+		},
+	}, nil
+}
+
+func createSubscription(ctx context.Context, topicName string, subscriptionName string, connectionString string, opts[] servicebus.SubscriptionManagementOption) (error) {
+	ns, err := getSbNamespace(connectionString)
+	if err != nil {
+		return err
+	}
+
+	subManager, err := ns.NewSubscriptionManager(topicName)
+	if err != nil {
+		return err
+	}
+	
+	_, err = ensureSubscription(ctx, subManager, subscriptionName, opts...)	
+	return err	
+}
+
+func ensureSubscription(ctx context.Context, sm *servicebus.SubscriptionManager, name string, opts ...servicebus.SubscriptionManagementOption) (*servicebus.SubscriptionEntity, error) {
+	subEntity, err := sm.Get(ctx, name)
+	if err == nil {
+		_ = sm.Delete(ctx, name)
+	}
+
+	subEntity, err = sm.Put(ctx, name, opts...)
+	if err != nil {		
+		return nil, err
+	}
+
+	return subEntity, nil
+}
+
+func deleteSubscription(ctx context.Context, topicName string, subscriptionName string, connectionString string) (error){
+	ns, err := getSbNamespace(connectionString)
+	if err != nil {
+		return err
+	}
+	sm, err := ns.NewSubscriptionManager(topicName)
+	se, err := sm.Get(ctx, subscriptionName)
+	if se != nil {
+		_ = sm.Delete(ctx, subscriptionName)
+	}
+	return nil
+}
+
+func (s *subscription) getSbSubscription(ctx context.Context)(*servicebus.Subscription, error){
+	sm, err := s.ns.NewSubscriptionManager(s.topicName)	
+	if err != nil {
+		return nil, err
+	}
+	// An empty SubscriptionEntity means no subscription exist for the given name. 
+	se, err := sm.Get(ctx, s.name)
+	if se == nil {
+		return nil, fmt.Errorf("azurepubsub: no such subscription %v", s.name)
+	}
+
+	sbTopic, err := s.ns.NewTopic(s.topicName)	
+	if err != nil {
+		return nil, err
+	}
+	return sbTopic.NewSubscription(s.name)
+}
+
+// IsRetryable implements driver.Subscription.IsRetryable.
+func (s *subscription) IsRetryable(error) bool {	
+	return false
+}
+
+// As implements driver.Subscription.As.
+func (s *subscription) As(i interface{}) bool {
+	p, ok := i.(**servicebus.Subscription)
+	if !ok {
+		return false
+	}	
+	*p, _ = s.getSbSubscription(context.Background())
```

Comment:
Why is the error ignored? Shouldn't it return false if `err != nil`?

Reply:
This was changed with my rework of OpenXX; the ServiceBus Subscription and Topic is now owned by the caller. Please look at the latest changes.